### PR TITLE
Add a test case for the SendJoinParser

### PR DESCRIPTION
**Pull request by DMRobertson**
This would have caught the bug #11438 introduced in #11217 and fixed in #11439. I confirmed this by `pip install ijson==3.0`, running the test and seeing a failure.

### CODE REVIEW

**Comment by danimo** [comment]
Thanks for the test!

File: changelog.d/11441.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Add a test case to sanity check that we parse `/send_join` responses correctly.
```
Comment (babolivier) [inline changelog.d/11441.misc:1]: Can we group it with the changelog from https://github.com/matrix-org/synapse/pull/11439?

**Review by babolivier — APPROVED** [review]
Thanks for taking care of it! Just a nitpick with the changelog but happy for it to be merged once resolved.

File: tests/federation/transport/test_client.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import json
```
Comment (clokep) [inline tests/federation/transport/test_client.py:1]: @DMRobertson Can you add a license header to this file please?

Comment (clokep) [reply tests/federation/transport/test_client.py:1]: See #11460.

**Pull request merged**